cli/internal/daemon/connector/connector.go

```diff
@@ -269,23 +270,29 @@ func (c *Connector) connectInternal(ctx context.Context) (*Client, error) {
 // the daemon if it doesn't find one running.
 func (c *Connector) getOrStartDaemon() (int, error) {
 	lockFile := c.lockFile()
-	if daemonProcess, err := lockFile.GetOwner(); errors.Is(err, lockfile.ErrDeadOwner) {
-		// If we've found a pid file but no corresponding process, there's nothing we can do.
-		// We defer to the user to clean up the pid file.
-		return 0, errors.Wrapf(err, "pid file appears stale. If no daemon is running, please remove it: %v", c.PidPath)
-	} else if os.IsNotExist(err) {
-		if c.Opts.DontStart {
-			return 0, ErrDaemonNotRunning
-		}
-		// The pid file doesn't exist. Start a daemon
-		pid, err := c.startDaemon()
-		if err != nil {
-			return 0, err
+	daemonProcess, getDaemonProcessErr := lockFile.GetOwner()
+	if getDaemonProcessErr != nil {
+		// If we're in a clean state this isn't an "error" per se.
+		// We attempt to start a daemon.
+		if errors.Is(getDaemonProcessErr, fs.ErrNotExist) {
+			if c.Opts.DontStart {
+				return 0, ErrDaemonNotRunning
+			}
+			pid, startDaemonErr := c.startDaemon()
+			if startDaemonErr != nil {
+				return 0, startDaemonErr
+			}
+			return pid, nil
 		}
-		return pid, nil
-	} else {
-		return daemonProcess.Pid, nil
+
+		// We could have hit any number of errors.
+		// - Failed to read the file for permission reasons.
+		// - User emptied the file's contents.
+		// - etc.
+		return 0, errors.Wrapf(getDaemonProcessErr, "An issue was encountered with the pid file. Please remove it and try again: %v", c.PidPath)
```

Comment:
Just for my Golang-chops... `%v` instead of `%s` here? `c.PidPath` is an `AbsoluteSystemPath` which is a `string`. Is there any reason to use one or the other? Why does go even give us so many options to interpolate?

Reply:
`%v` is _almost always_ a good choice so I pretty much only ever use it.

Reply:
They likely primarily exist because of path dependence on C's implementation.